industrial_ci/src/tests/source_tests.sh

```diff
@@ -139,43 +141,32 @@ if [ "$NOT_TEST_BUILD" != "true" ]; then
     ici_time_start catkin_run_tests
 
     if [ "$BUILDER" == catkin ]; then
-        source $CATKIN_WORKSPACE/devel/setup.bash # force to update ROS_PACKAGE_PATH for rostest
         catkin run_tests $OPT_VI --no-deps --no-status $PKGS_DOWNSTREAM $CATKIN_PARALLEL_TEST_JOBS --make-args $ROS_PARALLEL_TEST_JOBS --
         catkin_test_results $CATKIN_WORKSPACE || error
     fi
 
     ici_time_end  # catkin_run_tests
 fi
 
-
 if [ "$NOT_TEST_INSTALL" != "true" ]; then
 
-    ici_time_start catkin_install_build
-
-    # Test if the packages in the downstream repo build.
-    if [ "$BUILDER" == catkin ]; then
-        catkin clean --yes || catkin clean -b
-        catkin config --install
-        catkin build $OPT_VI --summarize --no-status $BUILD_PKGS_WHITELIST $CATKIN_PARALLEL_JOBS --make-args $ROS_PARALLEL_JOBS
-        source $CATKIN_WORKSPACE/install/setup.bash
-    fi
-
-    ici_time_end  # catkin_install_build
     ici_time_start catkin_install_run_tests
 
-    export EXIT_STATUS=0
+    EXIT_STATUS=0
     # Test if the unit tests in the packages in the downstream repo pass.
     if [ "$BUILDER" == catkin ]; then
       for pkg in $PKGS_DOWNSTREAM; do
+        if [ ! -d "$CATKIN_WORKSPACE/install/share/$pkg" ]; then continue; fi # skip meta-packages
```

Comment:
No, it just skips packages without a directory in `install/share`. AFAIK this is only the case for meta-packages.